USING_VANAGON.md

```diff
@@ -0,0 +1,42 @@
+
+### Components
+The process vanagon uses to actually build the MSI doesn’t break the normal path used for Unix based projects; vanagon builds each component on a build system, moves everything to a staging directory, then packages everything up.  In order to construct components on a windows system follow the usual vanagon pattern set out in the [styleguide](https://github.com/puppetlabs/vanagon/blob/master/STYLEGUIDE.md).
+
+Component files and project files likely require some differentiation based on installation between Unix and Windows. Common practice to deal with the differences is to use the is_"platform"? commands to create different branches of configuration. A list of the available is_"platform"? Functions can be found under [platform.rb](https://github.com/puppetlabs/vanagon/blob/master/lib/vanagon/platform.rb).
+
+Once the components successfully build on the build system, vanagon will tar up the combined source and then unpack it under /var/tmp/#{workdir}/SourceDir. For reference, SourceDir contains exactly what the install will look like from a component standpoint.  This is a good place to check for directory and file structure.  After unpacking the components to SourceDir vanagon uses the WiX compiler to create an MSI.
+
+### WiX Compiler
+Vanagon uses three parts of the WiX compiler toolset to generate, compile and link WiX files to create a final MSI. Information about the compilation and link steps aren’t specifically pertinent to the use of vanagon. Vanagon projects don’t require changes to those steps to work. Refer to the Vanagon specifications documentation for more info and WiX docs on the linker and compiler.
+
+Vanagon projects may require some knowledge of the generation step. WiX provides the “heat” tool to generate elements corresponding to every file and directory inside an input directory. Vanagon uses the heat tool to generate an “AppComponentGroup” element containing elements for every file and folder under the installation directory. If a project must create specific files/directories with specific attributes (such as a permanent file or folder) those files/directories need to be filtered out from the execution of heat.
+
+Filtering files from heat consists of using an xml filter as input to heat.exe. Heat automatically filters out anything that file matches to. Currently, vanagon contains a [filter](https://github.com/puppetlabs/vanagon/blob/master/resources/windows/wix/filter.xslt.erb) for service files. To filter out other files, move a copy of that filter to the project resources/windows/wix dir, then extend the filter to add the other files.
+
+Due to the appdata directory tendency to be special case (specifically the fact that files and directories there are usually permanent) heat will not run there. Thus the heat filter requires nothing extra for appdata files and directories.
```

Comment:
Rephrase - heat is not appropriate for this task - it is preferable to write out the xml directly.
Remember - Heat is just a tool that auto-generates the xml for a simple file structure package with no extra complexity.